src/ucp/core/ucp_ep.c

```diff
@@ -1622,13 +1668,21 @@ ucs_status_t ucp_ep_config_init(ucp_worker_h worker, ucp_ep_config_t *config,
         if (rsc_index != UCP_NULL_RESOURCE) {
             iface_attr = ucp_worker_iface_get_attr(worker, rsc_index);
             /* PUT */
+            if (iface_attr->cap.flags & UCT_IFACE_FLAG_PUT_SHORT) {
+                rma_config->max_put_short = iface_attr->cap.put.max_short;
+            } else {
+                rma_config->max_put_short = -1;
+            }
             if (iface_attr->cap.flags & UCT_IFACE_FLAG_PUT_ZCOPY) {
-                rma_config->max_put_zcopy    = iface_attr->cap.put.max_zcopy;
-                /* TODO: formula */
+                rma_config->max_put_zcopy = iface_attr->cap.put.max_zcopy;
                 if (context->config.ext.zcopy_thresh == UCS_MEMUNITS_AUTO) {
-                    rma_config->put_zcopy_thresh = 16384; 
+                    /* TODO: Use calculated value for PUT Zcopy threshold */
+                    rma_config->put_zcopy_thresh = 16384;
                 } else {
-                    rma_config->put_zcopy_thresh = context->config.ext.zcopy_thresh; 
+                    rma_config->put_zcopy_thresh = context->config.ext.zcopy_thresh;
+
+                    ucp_ep_config_adjust_max_short(&rma_config->max_put_short,
```

Comment:
in theory max_short can be large, so this adjustment is relevant for `UCS_MEMUNITS_AUTO` as well

Reply:
ok